src/dataflow/components/dataflow-program-toolbar.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { BaseComponent, IBaseProps } from "./dataflow-base";
+import { observer, inject } from "mobx-react";
+import { NodeTypes } from "../utilities/node";
+
+import "./dataflow-program-toolbar.sass";
+
+interface IProps extends IBaseProps {
+  onNodeCreateClick: (type: string) => void;
+  onDeleteClick: () => void;
+  onResetClick: () => void;
+  onClearClick: () => void;
+}
+interface IState {}
+
+@inject("stores")
+@observer
+export class DataflowProgramToolbar extends BaseComponent<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      showSettings: false,
```

Comment:
This does not appear to be used and `IState` is defined as empty above. If you were to define it as
```
this.state: IState = { showSettings: false };
```
I presume you'd get a compile-time type error.

Reply:
Right.  Fixed.